### PR TITLE
eg for the java generic tests (squashed)

### PR DESCRIPTION
**Pull request by Artem-applitools**

### CODE REVIEW

**Comment by Artem-applitools** [comment]
Current test run on the `eg_java` branch of the sdk.coverage.tests 
PR should be merged right after merge of https://github.com/applitools/sdk.coverage.tests/pull/99 
EG speed up generic tests although still require some tests to be run locally.

File: .travis.yml [hunk 1]
```diff
@@ -94,11 +94,24 @@ jobs:
...
+        firefox: latest
       script:
+        - "export DISPLAY=:99.0"
+        - "sh -e /etc/init.d/xvfb start"
+        - sleep 3
         - echo $APPLITOOLS_REPORT_ID
         - nvm install 14.5.0
         - nvm use 14.5.0
         - mvn clean install -DskipTests
+        - chmod +x ./initGeckoDriver.sh && ./initGeckoDriver.sh
```
Comment (ikalinin1) [inline .travis.yml:111]: Have you considered alternative solutions like https://github.com/bonigarcia/webdrivermanager or https://github.com/selenide/selenide ?

Comment (Artem-applitools) [reply .travis.yml:111]: it's script from another sdk, so logic will be more similar cross sdks here. 
The test written for the selenium, don't see why to migrate them to the selenide.
Webdrivermanager actually seems good. Although don't think it is better than a small script to add it as a dependency.

File: coverage-tests/configuration/eg_configuration.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+module.exports = {
+  name: 'eyes_selenium_java',
+  emitter: 'https://raw.githubusercontent.com/applitools/sdk.coverage.tests/master/java/emitter.js',
```
Comment (Yarden-Ingber) [inline coverage-tests/configuration/eg_configuration.js:3]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/applitools/eyes.sdk.java3/pullRequest?prid=7879168)
```suggestion
  emitter: "https://raw.githubusercontent.com/applitools/sdk.coverage.tests/master/java/emitter.js",
```

File: coverage-tests/configuration/eg_configuration.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+module.exports = {
+  name: 'eyes_selenium_java',
```
Comment (Yarden-Ingber) [inline coverage-tests/configuration/eg_configuration.js:2]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/applitools/eyes.sdk.java3/pullRequest?prid=7879168)
```suggestion
  name: "eyes_selenium_java",
```

File: coverage-tests/configuration/eg_configuration.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+  emitter: 'https://raw.githubusercontent.com/applitools/sdk.coverage.tests/master/java/emitter.js',
+  overrides: [
+    'https://raw.githubusercontent.com/applitools/sdk.coverage.tests/master/java/overrides.js',
+    'https://raw.githubusercontent.com/applitools/sdk.coverage.tests/master/eg.overrides.js'
+  ],
+  template: 'https://raw.githubusercontent.com/applitools/sdk.coverage.tests/master/java/template.hbs',
+  tests: 'https://raw.githubusercontent.com/applitools/sdk.coverage.tests/master/coverage-tests.js',
+  ext: '.java',
```
Comment (Yarden-Ingber) [inline coverage-tests/configuration/eg_configuration.js:10]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/applitools/eyes.sdk.java3/pullRequest?prid=7879168)
```suggestion
  ext: ".java",
```

File: coverage-tests/configuration/eg_configuration.js [hunk 5]
```diff
@@ -0,0 +1,12 @@
+module.exports = {
+  name: 'eyes_selenium_java',
+  emitter: 'https://raw.githubusercontent.com/applitools/sdk.coverage.tests/master/java/emitter.js',
+  overrides: [
+    'https://raw.githubusercontent.com/applitools/sdk.coverage.tests/master/java/overrides.js',
+    'https://raw.githubusercontent.com/applitools/sdk.coverage.tests/master/eg.overrides.js'
+  ],
+  template: 'https://raw.githubusercontent.com/applitools/sdk.coverage.tests/master/java/template.hbs',
```
Comment (Yarden-Ingber) [inline coverage-tests/configuration/eg_configuration.js:8]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/applitools/eyes.sdk.java3/pullRequest?prid=7879168)
```suggestion
  template: "https://raw.githubusercontent.com/applitools/sdk.coverage.tests/master/java/template.hbs",
```

File: coverage-tests/configuration/eg_configuration.js [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+  tests: 'https://raw.githubusercontent.com/applitools/sdk.coverage.tests/master/coverage-tests.js',
```
Comment (Yarden-Ingber) [inline coverage-tests/configuration/eg_configuration.js:9]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/applitools/eyes.sdk.java3/pullRequest?prid=7879168)
```suggestion
  tests: "https://raw.githubusercontent.com/applitools/sdk.coverage.tests/master/coverage-tests.js",
```

File: coverage-tests/configuration/eg_configuration.js [hunk 7]
```diff
@@ -0,0 +1,12 @@
+module.exports = {
+  name: 'eyes_selenium_java',
+  emitter: 'https://raw.githubusercontent.com/applitools/sdk.coverage.tests/master/java/emitter.js',
+  overrides: [
+    'https://raw.githubusercontent.com/applitools/sdk.coverage.tests/master/java/overrides.js',
+    'https://raw.githubusercontent.com/applitools/sdk.coverage.tests/master/eg.overrides.js'
```
Comment (Yarden-Ingber) [inline coverage-tests/configuration/eg_configuration.js:6]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/applitools/eyes.sdk.java3/pullRequest?prid=7879168)
```suggestion
    "https://raw.githubusercontent.com/applitools/sdk.coverage.tests/master/eg.overrides.js"
```

File: coverage-tests/configuration/eg_configuration.js [hunk 8]
```diff
@@ -0,0 +1,12 @@
...
+  ],
+  template: 'https://raw.githubusercontent.com/applitools/sdk.coverage.tests/master/java/template.hbs',
+  tests: 'https://raw.githubusercontent.com/applitools/sdk.coverage.tests/master/coverage-tests.js',
+  ext: '.java',
+  outPath: './src/test/java/coverage/generic',
```
Comment (Yarden-Ingber) [inline coverage-tests/configuration/eg_configuration.js:11]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/applitools/eyes.sdk.java3/pullRequest?prid=7879168)
```suggestion
  outPath: "./src/test/java/coverage/generic",
```

File: coverage-tests/configuration/eg_configuration.js [hunk 9]
```diff
@@ -0,0 +1,12 @@
+module.exports = {
+  name: 'eyes_selenium_java',
+  emitter: 'https://raw.githubusercontent.com/applitools/sdk.coverage.tests/master/java/emitter.js',
+  overrides: [
+    'https://raw.githubusercontent.com/applitools/sdk.coverage.tests/master/java/overrides.js',
```
Comment (Yarden-Ingber) [inline coverage-tests/configuration/eg_configuration.js:5]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/applitools/eyes.sdk.java3/pullRequest?prid=7879168)
```suggestion
    "https://raw.githubusercontent.com/applitools/sdk.coverage.tests/master/java/overrides.js",
```

File: coverage-tests/configuration/index.js [hunk 10]
```diff
@@ -1,8 +1,11 @@
 module.exports = {
   name: 'eyes_selenium_java',
   emitter: 'https://raw.githubusercontent.com/applitools/sdk.coverage.tests/master/java/emitter.js',
-  overrides: 'https://raw.githubusercontent.com/applitools/sdk.coverage.tests/master/java/overrides.js',
+  overrides: [
+    'https://raw.githubusercontent.com/applitools/sdk.coverage.tests/master/java/overrides.js',
+  ],
   template: 'https://raw.githubusercontent.com/applitools/sdk.coverage.tests/master/java/template.hbs',
+  tests: 'https://raw.githubusercontent.com/applitools/sdk.coverage.tests/master/coverage-tests.js',
```
Comment (Yarden-Ingber) [inline coverage-tests/configuration/index.js:8]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/applitools/eyes.sdk.java3/pullRequest?prid=7879168)
```suggestion
  tests: "https://raw.githubusercontent.com/applitools/sdk.coverage.tests/master/coverage-tests.js",
```

Comment (Yarden-Ingber) [inline coverage-tests/configuration/index.js:8]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/applitools/eyes.sdk.java3/pullRequest?prid=7879168)
```suggestion
  tests: "https://raw.githubusercontent.com/applitools/sdk.coverage.tests/master/coverage-tests.js",
```

File: coverage-tests/configuration/index.js [hunk 11]
```diff
@@ -1,8 +1,11 @@
 module.exports = {
   name: 'eyes_selenium_java',
   emitter: 'https://raw.githubusercontent.com/applitools/sdk.coverage.tests/master/java/emitter.js',
-  overrides: 'https://raw.githubusercontent.com/applitools/sdk.coverage.tests/master/java/overrides.js',
+  overrides: [
+    'https://raw.githubusercontent.com/applitools/sdk.coverage.tests/master/java/overrides.js',
```
Comment (Yarden-Ingber) [inline coverage-tests/configuration/index.js:5]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/applitools/eyes.sdk.java3/pullRequest?prid=7879168)
```suggestion
    "https://raw.githubusercontent.com/applitools/sdk.coverage.tests/master/java/overrides.js",
```

**Pull request merged**